src/main/webapp/parse.js

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/** 
+ * parse.js
+ *
+ * MediaWiki API Demos
+ * Demo of `Parse` module: Parse content of a page
+ *
+ * MIT License
+ */
+
+function sendData() {
+    var url;
+    try {
+        url = "https://en.wikibooks.org/w/api.php?" +
+        new URLSearchParams({
+            origin: "*",
+            action: "parse",
+            page: `Cookbook:${document.getElementById("input-text").value}`,
+            format: "json",
+            prop: "text",
```

Comment:
Consider the following for a future pull request:
"text" returns the parsed HTML page - you take advantage of it by loading the HTML and navigating it using `document.getElementsBy*`.
However, did you consider using "wikitext" instead? It may not have an HTML format, but it's simpler to analyze, to the point that you can process the article by searching the text with `.match()` function (for such keywords as "==Ingredients==").
An example of a wikitext response is [here](https://en.wikibooks.org/w/api.php?action=parse&page=Cookbook:Racuszki&format=json&prop=wikitext).

Reply:
The wikitext format requires additional processing like remove extra symbols, for example if "batter" is a link and it represented like 
[[Cookbook:Batter|batter]]
Thats why I use text, however some HTML tags on wiki pages don't have ElementId, and I should check if webpage has correct structure.